test/unit/Cardano/Wallet/BinarySpec.hs

```diff
@@ -0,0 +1,118 @@
+
+import qualified Data.Set                         as Set
+
+import           Data.ByteArray.Encoding          (convertFromBase, Base (Base16))
+import           Data.ByteString                  (ByteString)
+import qualified Data.ByteString.Lazy.Char8 as L8
+import           Data.ByteString.Base58           (bitcoinAlphabet, decodeBase58)
+
+import Test.Hspec
+    ( Spec, describe, it, shouldBe )
+
+import Cardano.Wallet.Primitive
+import Cardano.Wallet.Binary.Helpers (unsafeDeserialiseFromBytes)
+import Cardano.Wallet.Binary (decodeBlock, decodeBlockHeader)
+
+-- A mainnet block
+block1 :: Block
+block1 = Block
+    { header = BlockHeader
+        { epochIndex = 105
+        , slotNumber = 9519
+        , prevBlockHash = hash16 "4d97da40fb62bec847d6123762e82f9325f11d0c8e89deee0c7dbb598ed5f0cf"
+        }
+    , transactions = mempty
+    }
+
+-- A mainnet block with a transaction
+block2 :: Block
+block2 = Block
+    { header = BlockHeader
+        { epochIndex = 105
+        , slotNumber = 9876
+        , prevBlockHash = hash16 "da73001193ab3e6a43921385941c5f96b8f56de3908e78fae06f038b91dadd9d"
+        }
+    , transactions = Set.fromList
+        [ Tx { inputs =
+               [ TxIn { txId = hash16 "60dbb2679ee920540c18195a3d92ee9be50aee6ed5f891d92d51db8a76b02cd2"
+                      , txIx = 3} ]
+             , outputs =
+               [ TxOut { address = addr58 "DdzFFzCqrhsug8jKBMV5Cr94hKY4DrbJtkUpqptoGEkovR2QSkcAcRgjnUyegE689qBX6b2kyxyNvCL6mfqiarzRB9TRq8zwJphR31pr"
+                       , coin = Coin 285000000
+                       }
+               , TxOut { address = addr58 "DdzFFzCqrhsmxmuQpgjUrvRwF5ZKnyQ7pGrS4q53u5B516wcc26maHz9M4myYAkQVc5m9E4DKJjRDjPxuDdK3ZsHb1Dnqf3XorZ1PnzX"
+                       , coin = Coin 1810771919
+                       }
+               ]
+             } ]
+    }
+
+-- A testnet block with a transaction
+block3 :: Block
+block3 = Block
+    { header = BlockHeader
+        { epochIndex = 30
+        , slotNumber = 9278
+        , prevBlockHash = hash16 "b065b5fe97bec5fd130e7a639189499c9d0b1fcf9348c5c19f7a22700da7a35e"
+        }
+    , transactions = Set.fromList
+        [ Tx { inputs =
+               [ TxIn { txId = hash16 "6967e2b5c3ad5ae07a9bd8d888f1836195a04f7a1cb4b6d083261870068fab1b"
+                      , txIx = 1}
+               , TxIn { txId = hash16 "7064addc0968bccd7d57d2e7aa1e9c2f666d8387042483fc1e87200cfb96c8f1"
+                      , txIx = 0} ]
+             , outputs =
+               [ TxOut { address = addr58 "37btjrVyb4KBsw2f3V76ntfwqDPgyf3QmmdsrTSmCnuTGYtS9JgVXzxeQEsKjgWurKoyw9BDNEtLxWtU9znK49SC8bLTirk6YqcAESFxXJkSyXhQKL"
+                       , coin = Coin 1404176490
+                       }
+               , TxOut { address = addr58 "37btjrVyb4KD5Ne4yvGAHGbQuHUYQX1VPsXh85rBh3UrGSMWdRSFxBYQ9RQRHCMezN6AMLd3uYTC5hbeVTUiPzfQUTCEogg2HrSJKQUjAgsoYZHwT3"
+                       , coin = Coin 1004099328
+                       }
+               ]
+             } ]
+    }
```

Comment:
Same as above about the identation (cf: https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#avoid-variable-length-indentation)